src/mongo/shell.ts

```diff
@@ -72,43 +88,59 @@ export class Shell {
 	async exec(script: string): Promise<string> {
 		script = this.convertToSingleLine(script);
 		const executionId = this._generateExecutionSequenceId();
+		// tslint:disable-next-line:no-any
+		let writeError: any;
 
 		try {
 			this.mongoShell.stdin.write(script, 'utf8');
 			this.mongoShell.stdin.write(os.EOL);
 			this.mongoShell.stdin.write(executionId, 'utf8');
 			this.mongoShell.stdin.write(os.EOL);
 		} catch (error) {
-			window.showErrorMessage(error.toString());
+			// Generally if writing to the process' stdin fails it has already exited
+			// with an error, and we will get notification via its stdout. So hold on to
+			// this and only return it if we don't see any other notifications after timeout.
+			writeError = error;
```

Comment:
Can you give an example of what this error would look like? I'm just confused why we would need to wait for the timeout to occur before throwing it.